omni-relayer/src/workers/near.rs

```diff
@@ -0,0 +1,255 @@
+        ))
+        .await;
+    }
+}
+
+pub async fn finalize_transfer(
+    config: config::Config,
+    redis_client: redis::Client,
+    connector: Arc<Nep141Connector>,
+) -> Result<()> {
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection_clone = redis_connection.clone();
+        let Some(mut events) = utils::redis::get_events(
+            &mut redis_connection_clone,
+            config.redis.near_sign_transfer_events.clone(),
+        )
+        .await
+        else {
+            tokio::time::sleep(tokio::time::Duration::from_secs(
+                config.redis.sleep_time_after_events_process_secs,
+            ))
+            .await;
+            continue;
+        };
+
+        let mut handlers = Vec::new();
+        while let Some((nonce, event)) = events.next_item().await {
+            if let Ok(event) = serde_json::from_str::<Nep141LockerEvent>(&event) {
+                handlers.push(tokio::spawn({
+                    let config = config.clone();
+                    let mut redis_connection = redis_connection.clone();
+                    let connector = connector.clone();
+
+                    async move {
+                        let Nep141LockerEvent::SignTransferEvent { .. } = &event else {
+                            error!("Expected SignTransferEvent, got: {:?}", event);
+                            return;
+                        };
+
+                        match connector.finalize_deposit_omni_with_log(event).await {
+                            Ok(tx_hash) => {
+                                info!("Finalized deposit: {}", tx_hash);
+                                utils::redis::remove_event(
+                                    &mut redis_connection,
+                                    &config.redis.near_sign_transfer_events,
+                                    &nonce,
+                                )
+                                .await;
+                            }
+                            Err(err) => {
+                                error!("Failed to finalize deposit: {}", err);
+                            }
+                        }
+                    }
+                }));
+            }
+        }
+
+        join_all(handlers).await;
+
+        tokio::time::sleep(tokio::time::Duration::from_secs(
+            config.redis.sleep_time_after_events_process_secs,
+        ))
+        .await;
+    }
+}
+
+pub async fn claim_fee(
+    config: config::Config,
+    redis_client: redis::Client,
+    connector: Arc<Nep141Connector>,
+) -> Result<()> {
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection_clone = redis_connection.clone();
+        let Some(mut events) = utils::redis::get_events(
+            &mut redis_connection_clone,
+            config.redis.eth_deposit_events.clone(),
```

Comment:
These don't have to be Ethereum events. In the future, they could be events from Solana, for example. In this file, interactions with blockchains other than NEAR should be encapsulated. Maybe this function should be moved to eth.rs.

Reply:
For now I renamed it to "finalized_transfers", but in the future I'd need to pass simple `Deposit` structure instead of `Log<customDeposit>`. I can't do it for now, because it expects only `eth` proof for now. I need to think how to refactor it. Right now I'm thinking about storing directly `ClaimFeeArgs` in redis instead of event structure

Reply:
Done. It was fixed by storing `FinTransferArgs` and `ClaimTransferArgs` in db, instead of storing custom eth log structure. However, I'm not sure if we should move `claim_fee` method in eth.rs worker, claim_fee feels more attached to token locker account and near blockchain itself, but let me know if you still think that it'd be better to move `claim_fee` to eth.rs